### PR TITLE
[TEST-MERGE ONLY] Ports multiple optimizatioin PRs from liberty to hopefully reduce mob SS overhead

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request
https://github.com/Liberty-Landing/Liberty-Station-13/pull/446 Was the main port, but we tried it in https://github.com/sojourn-13/sojourn-station/pull/4697 and gave up beacuse it was messed up

There were follow up prs: https://github.com/Liberty-Landing/Liberty-Station-13/pull/455, https://github.com/Liberty-Landing/Liberty-Station-13/pull/453, https://github.com/Liberty-Landing/Liberty-Station-13/pull/485 that all fixed issues with it. We should give a shot, as the author has procalimed the modern version is far less buggy and more performant.

I have not throughly tested this, as I dont have much time, but I think we should TM this just to see how it goes.

## Changelog
:cl:
refactor: Mob AI is now on the moveloop instead of life loop
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -494,13 +494,17 @@ default behaviour is:
 	if (buckled)
 		return
 
+	// Anything that observes us moving, try to activate.
+	for(var/mob/living/M in oviewers(world.view, src))
```
Comment (nikothedude) [inline code/modules/mob/living/living.dm:498]: note, this is alledgedly a buff to mob ai, as it makes have the same vision as a player (which was apparantly not the case before)?

Comment (Yatiyaya) [reply code/modules/mob/living/living.dm:498]: TLDR, mobs would make a list() for all other mobs on the zlevel it occupies, and would run through it to check if it should activate (Based on faction/distance from it)

This change means that anytime a living entity moves within a mob's vision (Mob A moves, mobs B sees A move, so List(B) is generated on mob A)

**Review by Yatiyaya — CHANGES_REQUESTED** [review]
I hope this actually shows all the changes...

**Comment by Yatiyaya** [comment]
So, I was looking over this code and realized it didn't include a lot of functionally important things (And also did not include removal of specific parts)

**Comment by nikothedude** [comment]
well what wasnt removed

**Comment by nikothedude** [comment]
> well what wasnt removed

or wasnt added or wahtever

**Comment by Yatiyaya** [comment]
I would show, but right now I'm doing some spring cleaning in the mob AI code to get it more good

**Comment by Yatiyaya** [comment]
Because I don't know how to use github, here you go https://github.com/sojourn-13/sojourn-station/pull/5174

**Comment by nikothedude** [comment]
sure

**Pull request closed without merging**